File: DSA/LakshaKundra_590013237/Day_19_Question_1.py
def search_rotated(arr, target):
    low, high = 0, len(arr) - 1
    while low <= high:
        mid = (low + high) // 2
        if arr[mid] == target:
            return mid
        if arr[low] <= arr[mid]:
            if arr[low] <= target < arr[mid]:
                high = mid - 1
            else:
                low = mid + 1
        else:
            if arr[mid] < target <= arr[high]:
                low = mid + 1
            else:
                high = mid - 1
    return -1

print(search_rotated([4, 5, 6, 7, 0, 1, 2], 0))
print(search_rotated([4, 5, 6, 7, 0, 1, 2], 3))
print(search_rotated([11, 13, 15, 17, 19, 2, 3, 6, 7], 6))
  

File: DSA/Stuti_Jain_590012149/Day_19_Question_1.c
#include <stdio.h>
int search(int arr[], int n, int target){
    int low=0, high=n-1;
    while(low<=high){
        int mid=low+(high-low)/2;
        if(arr[mid]==target)
            return mid;
        if(arr[low]<=arr[mid]){
            if(target>=arr[low] && target<arr[mid])
                high=mid-1;
            else
                low=mid+1;
        } else {
            if(target>arr[mid] && target<=arr[high])
                low=mid+1;
            else
                high=mid-1;
        }
    }
    return -1;
}

int main(){
    int n, target;
    printf("Enter the size of array: ");
    scanf("%d", &n);
    int arr[n];
    printf("Enter the elements: ", n);
    for(int i=0; i<n; i++)
        scanf("%d", &arr[i]);
    printf("Target:");
    scanf("%d", &target);
    int index=search(arr,n,target);
    if(index!=-1)
        printf("%d", index);
    else
        printf("Target not found in the array",target);
    return 0;
}
